tests/integration/testlib/assertlib.py

```diff
@@ -35,13 +36,38 @@ def assert_state(desired_state_data):
     assert desired_state.state == current_state.state
 
 
-def assert_absent(*ifnames):
+def assert_absent(*ifnames, timeout=0):
     """ Assert that a interface is not present in the current state """
 
-    current_state = statelib.show_only(ifnames)
+    now = time.time()
+    while True:
```

Comment:
I would prefer the retry part be separated from the assert.
It can then be shared between asserts.

I can share one example of this from the VDSM project:
This is a decorator that can be used on an assert helper.
```
def retry_assert(func):
    def retry(*args, **kwargs):
        for i in range(4):
            try:
                func(*args, **kwargs)
            except AssertionError:
                time.sleep(1)
            else:
                return
        func(*args, **kwargs)

    return retry
```